miniscope_io/bit_operation.py

```diff
@@ -0,0 +1,93 @@
+"""
+Bit operations for parsing header and payload information. Currently for use in streamDaq module
+"""
+
+from typing import Tuple
+
+import numpy as np
+
+
+class BufferFormatter:
```

Comment:
> is this something that we think will be reused by other devices?

I think this is a yes because these are pretty generic bit operations. Dealing with bit-order stuff in embedded systems can get a bit confusing, so it would be useful to have these when figuring out data structure when debugging, etc. I agree that there's a decent number of top modules so moving this class into `utils` might make sense.

> could we have some more documentation here?

Yes, for sure. I'll put the example I wrote in the comments below, and write a bit more about the buffer structure and the conversion going on here.